microceph/cmd/microceph/cluster_remove.go

```diff
@@ -36,15 +41,123 @@ func (c *cmdClusterRemove) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	client, err := m.LocalClient()
+	cli, err := m.LocalClient()
 	if err != nil {
 		return err
 	}
 
-	err = client.DeleteClusterMember(context.Background(), args[0], c.flagForce)
+	return removeNode(cli, args[0], c.flagForce)
+}
+
+func removeNode(cli *microCli.Client, node string, force bool) error {
+
+	logger.Debugf("Removing cluster member %v, force: %v", node, force)
+
+	// check prerquisites unless we're forcing
+	if !force {
+		ok, err := checkPrerequisites(cli, node)
+		if err != nil {
+			return fmt.Errorf("Error checking prereqs: %v", err)
+		}
+		if !ok {
+			return fmt.Errorf("Prerequisites not met, not removing: %v", err)
+		}
+	}
+
+	// delete from ceph
+	err := deleteNodeServices(cli, node)
+	if err != nil {
+		// forcing makes errs non-fatal
+		if !force {
+			return err
+		}
+		logger.Warnf("Error deleting services from node %v: %v", node, err)
+	}
+
+	// delete from cluster db
+	err = client.MClient.DeleteClusterMember(cli, node, force)
+	logger.Debugf("DeleteClusterMember %v: %v", node, err)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func checkPrerequisites(cli *microCli.Client, name string) (bool, error) {
```

Comment:
The boolean returned by this function isn't too useful, imo. The returned error already provides the needed information, and checking for both the error _and_ the bool makes things more complex than they should.

Reply:
Ack